sematic/plugins/building/docker_builder.py

```diff
@@ -339,30 +388,170 @@ def _find_build_config_files(script_path: str) -> List[str]:
+    if build_config.image_script is not None:
+        effective_base_uri = _execute_build_script(
+            target=target, image_script=build_config.image_script
+        )
+
+    # appease mypy
+    assert effective_base_uri is not None
+
+    return _build_image_from_base(
+        target=target,
+        effective_base_uri=effective_base_uri,
+        build_config=build_config,
+        docker_client=docker_client,
+    )
+
+
+def _build_image_from_base(
+    target: str,
+    effective_base_uri: ImageURI,
+    build_config: BuildConfig,
+    docker_client: docker.DockerClient,  # type: ignore
+) -> Tuple[Image, ImageURI]:
+    """
+    Builds the container image to use by adding layers to an existing base image.
+    """
+    logger.info("Building image starting from base: %s", effective_base_uri)
+
+    dockerfile_contents = _generate_dockerfile_contents(
+        base_uri=effective_base_uri,
+        source_build_config=build_config.build,
+        target=target if build_config.base_uri is not None else None,
+    )
+    logger.info("Using Dockerfile:\n%s", dockerfile_contents)
+
+    # we have to create a tmp dockerfile and pass it instead of using the `fileobj` option
+    # of the docker_client, because it does not work with contexts, so operations like
+    # COPY do not work, as there exists no working dir context to copy them from
+    # API: https://docker-py.readthedocs.io/en/stable/api.html#module-docker.api.build
+    # Reported unresolved issue: https://github.com/docker/docker-py/issues/2105
+    with tempfile.NamedTemporaryFile(mode="wt", delete=True) as dockerfile:
+        dockerfile.write(dockerfile_contents)
+        dockerfile.flush()
+
+        optional_kwargs = dict()
+        if build_config.build is not None and build_config.build.platform is not None:
+            optional_kwargs["platform"] = build_config.build.platform
+
+        status_updates = docker_client.api.build(
+            dockerfile=dockerfile.name,
+            # use the project root as the context
+            # TODO: switch from project-relative paths to build file-relative paths
+            path=os.getcwd(),
+            tag=str(effective_base_uri),
+            decode=True,
+            **optional_kwargs,
         )
 
+    if status_updates is None:
+        logger.warning("Built image '%s' without any response", effective_base_uri)
+        image = docker_client.images.get(str(effective_base_uri))
+        return image, ImageURI.from_image(image)
+
+    for status_update in status_updates:
+        if "error" in status_update:
+            logger.error(
+                "Image build error details: '%s'", str(status_update.get("errorDetail"))
+            )
+            raise BuildError(f"Unable to build image: {status_update['error']}")
+
+        update_str = " ".join(sorted([f"{k}={v}" for k, v in status_update.items()]))
+        logger.info("Image build update: %s", update_str.strip())
```

Comment:
`debug`

Reply:
For this and for "Image push update", I've eliminated empty update logs.
Also, these only log a lot of lines when building on a clean env, without any locally-cached layers. I've added a section at the bottom of the PR description showing what this would look like.
The Bazel build system also prints a comparative number of lines, only that it replaces them instead of appending to them. Docker does the same, so the end result seems less verbose.
PTAL.